### ISSUE TITLE
Create a bot to check for license at top of each file

### ISSUE THREAD

**Issue by BradSwain** [coderrect-inc/OpenRace#22]
The AGPL license suggests including a header at the top of each file.

Ideally we should have a GitHub Action that can be run on each PR to check that the license header is at the top of all of our code files (.h/.cpp/.c), and the ability to exclude some directories that are licensed differently (some of the folders in the tests directory are under a different license).

Eachl Coderrect Inc. source file should start with the following text

```
/* Copyright 2021 Coderrect Inc. All Rights Reserved.
Licensed under the GNU Affero General Public License, version 3 or later (“AGPL”), as published by the Free Software Foundation.
You may not use this file except in compliance with the License.
You may obtain a copy of the License at
https://www.gnu.org/licenses/agpl-3.0.en.html
Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an “AS IS” BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
==============================================================================*/
```

The action should:
 - Run on PRs
 - Check that the license header is at the top of changed files
 - Skip checking on non-code files
 - Be configurable to skip checking on some directories

**Comment by leonplust** [comment]
Do you think we can use [this](https://github.com/kt3k/deno_license_checker) tool to check the licenses? This tool seems to meet all the requirements. If so, I would like to try to configurate that.

**Comment by BradSwain** [comment]
This tool looks like exactly what we need. If you would like to configure it and make a PR that would be great!
Thanks for the contribution! If you run into any problems or have any questions I am happy to help.

**Comment by leonplust** [comment]
Sorry for the late response, I was caught up with something else and forgot about this. I have created a PR for this.

**Comment by BradSwain** [comment]
Thanks! I will take a look at the PR.

> Sorry for the late response

No worries! We appreciate you volunteering to help us out.

**Comment by BradSwain** [comment]
closed by #89 

Thanks again.
